packages/runtime/src/controls/rich-text.ts

```diff
@@ -0,0 +1,132 @@
+import { IndexSignatureHack } from '../prop-controllers/descriptors'
+import type * as Slate from 'slate'
+import { BuilderEditMode } from '../utils/constants'
+import { PropController } from '../prop-controllers/instances'
+import { OnChangeParam, Editor } from 'slate-react'
+import { BoxModel } from '../box-model'
+
+export type RichTextControlData = IndexSignatureHack<Slate.ValueJSON>
+
+export const RichTextControlType = 'makeswift::controls::rich-text'
+
+type RichTextControlConfig = {
+  label?: string
+  defaultValue?: RichTextControlData
```

Comment:
I would like us not to leak the internal data structure in the public API. Since the prop is received as a `ReactNode`, this default value is really the only place where you have to think in terms of Slate data. Besides it exposing complexity it also makes it harder for us to change internals, like upgrading Slate and changing our data structure since it would be a public API change.

My suggestion would be to allow plaintext as a default value that we then convert to the data structure. For example, for the color control we support color strings even though the underlying data is a Makeswift swatch ID. We could also simply drop the default value option.

Reply:
In my mind, I was thinking "I will handle this in milestone 3," but starting out with a basic text default value means that we don't have to have a breaking change. Will update!